foundry.toml

```diff
@@ -11,6 +11,7 @@ fs_permissions = [
     { access = "read", path = "./test/utils/config/" },
     { access = "read", path = "./test/utils/payload/" },
 ]
+via-ir = true
```

Comment:
Please do not just embed this kind of change in an unrelated commit. Open a PR to list reasons why you want/support this change and start a discussion there.

Please answer the following:
- How much does it impact compile time?
- Would contract verification still work in `tokenlon-contract-private` repo? [ref](https://twitter.com/PaulRBerg/status/1737169159052468479)
- Would it impact the scripts we run in `tokenlon-contract-private` repo? [ref](https://hardhat.org/hardhat-runner/docs/reference/solidity-support#support-for-ir-based-codegen): `> Since Hardhat works much better when the optimizer is disabled, we don't completely support the viaIR option yet.`
- Are there any other caveats of enabling IR? These caveats need to be documented in readme before we enable this change.

Reply:
This does not mean I'm against enabling `via-ir` but rather we need more study and feedbacks before we enabling this kind of change. Otherwise we might waste more time debugging in the future.